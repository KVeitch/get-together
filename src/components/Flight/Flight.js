import React from 'react'
import './Flight.scss'

const Flight = ({ flight }) => {
  const {depart, arrival, from, to, arrivalTime, departTime, flights} = flight
  return (
    <section className="section__flight">
      <h3>{from} &rarr; {to}</h3>
      <p>Departs on {depart} at {departTime}</p>
      <p>Arrives on {arrival} at {arrivalTime}</p>
      <p>Flight {flights}</p>
    </section>
  )
}

export default Flight
