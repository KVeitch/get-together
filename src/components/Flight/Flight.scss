@import '../../variables.scss';

.section__flight {
  margin:10px 10px;
  border:2px solid $main-highlight;
  text-overflow: ellipsis;
  padding:10px;
  border-radius: 10px;

}