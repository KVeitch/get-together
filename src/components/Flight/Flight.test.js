import React from 'react';
import { shallow } from 'enzyme';
import Flight from './Flight';


describe('Flight', () => {
  const mockFlight = {
    arrival: '01/15/2020',
    arrivalTime: '1 : 05 PM',
    depart: '01/12/2020',
    departTime: '7 : 51 PM',
    from: 'DEN',
    to: 'MRY',
    flights: 'AA 420 and AA 5718'
  };
  const wrapper = shallow(<Flight flight={mockFlight} />);

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
