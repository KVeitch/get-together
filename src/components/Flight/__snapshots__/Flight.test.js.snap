// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="section__flight"
>
  <h3>
    DEN
     â†’ 
    MRY
  </h3>
  <p>
    Departs on 
    01/12/2020
     at 
    7 : 51 PM
  </p>
  <p>
    Arrives on 
    01/15/2020
     at 
    1 : 05 PM
  </p>
  <p>
    Flight 
    AA 420 and AA 5718
  </p>
</section>
`;
